src/main/java/taxi/dao/DriverDaoImpl.java

```diff
@@ -111,4 +115,24 @@ private Driver parseDriverFromResultSet(ResultSet resultSet) throws SQLException
         driver.setLicenseNumber(licenseNumber);
         return driver;
     }
+
+    @Override
+    public Optional<Driver> findByLogin(String login) {
+        String query = "SELECT id, name, license_number, login, password FROM drivers"
+                + " WHERE login = ? AND is_deleted = FALSE;";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement preparedStatement = connection.prepareStatement(query)) {
+            preparedStatement.setString(1,login);
+            ResultSet resultSet = preparedStatement.executeQuery();
+            Driver driver = null;
+            if (resultSet.next()) {
+                driver = parseDriverFromResultSet(resultSet);
+                driver.setLogin(resultSet.getString("login"));
+                driver.setPassword(resultSet.getString("password"));
```

Comment:
lets add this logic into parseDriverFromResultSet() method (I think we should get all info about driver in all methods)